### DISCUSSION TITLE
Looking for a little help

### DISCUSSION THREAD

**Discussion by ibrunops** [bengtmartensson/IrScrutinizer#467] [Export issues] [answered]
Hi! I'm trying to create some Python code for a specific functionality I need. Basically, I need to convert an IR pulse frequency to Pronto Hex format. IrScrutinizer does it beautifully, I just copy the frequency, export it to Pronto Hex Oneshot and it gives me the output I need. But unfortunately, and after spending hours searching the web, I can't find what algorithm is used to perform this conversion. I tried many methods, even got "similar" results, but "similar" is not enough in programming world, and I can't find the solution. I know the frequency is 38000 Hz, I don't know if it has something to do with it. Honestly, I'm just getting into the world of IR signals.

For example, here are some correct inputs/outputs (the inputs are in RAW as I indicated above, and the output is in Pronto Hex:

**Input:** `700 700 700 2100 1400 700 700 2800 700 1400 700 700 700 1400 1400 700 700 1400 700 700 700 700 700 700 700 2100 700 2100 700 700 1400 700 700 2100 700 2800 1400 2800 700`
**Output:** `0000 006D 0014 0000 001B 001B 001B 0050 0035 001B 001B 006A 001B 0035 001B 001B 001B 0035 0035 001B 001B 0035 001B 001B 001B 001B 001B 001B 001B 0050 001B 0050 001B 001B 0035 001B 001B 0050 001B 006A 0035 006A 001B 076C`

**Input:** `1400 1400 700 700 700 700 1400 2800 700 2100 700 700 700 1400 700 1400 1400 2800 1400 2800 700 2800 1400 1400 700 2100 700 700 700 1400 1400 2800 700`
**Output:** `0000 006D 0011 0000 0035 0035 001B 001B 001B 001B 0035 006A 001B 0050 001B 001B 001B 0035 001B 0035 0035 006A 0035 006A 001B 006A 0035 0035 001B 0050 001B 001B 001B 0035 0035 006A 001B 076C`

**Input:** `700 700 700 700 1400 1400 1400 2800 700 1400 700 700 1400 1400 700 1400 1400 2800 700 700 700 2100 700 1400 700 1400 1400 700 700 700 700 700 700 2800 1400 2800 700`
**Output:** `0000 006D 0013 0000 001B 001B 001B 001B 0035 0035 0035 006A 001B 0035 001B 001B 0035 0035 001B 0035 0035 006A 001B 001B 001B 0050 001B 0035 001B 0035 0035 001B 001B 001B 001B 001B 001B 006A 0035 006A 001B 076C`

**Input:** `700 700 700 2100 1400 700 700 2800 700 1400 700 700 700 2800 1400 1400 700 2100 700 1400 700 1400 700`
**Output:** `0000 006D 000C 0000 001B 001B 001B 0050 0035 001B 001B 006A 001B 0035 001B 001B 001B 006A 0035 0035 001B 0050 001B 0035 001B 0035 001B 076C`

If anyone can give me any clues or solutions, I would greatly appreciate it. I don't know if this query is allowed, I apologize in advance.
Thanks!

**Comment by bengtmartensson** [comment, marked as answer]
Possibly [this](http://www.harctoolbox.org/Glossary.html#ProntoSemantics) answers your question? It should be an easy exercise to turn it into, e.g., Python.

**Comment by ibrunops** [comment]
Thanks for the reply! I had already read that guide, but I'm having trouble
understanding the 3rd and 4th number. I understand that they have to do
with the number of pairs (= half the number of durations) in the start- and
the repeat sequence respectively. But I can't figure out what they mean by
durations and how to identify the start- sequence and the repeat sequence.
In my examples, the third number always changes, while the fourth is always
0000, so I understand there is no "repeat sequence" ever. But I can't
understand the reason for those two numbers and how to get them, if you can
give me any clue I'll appreciate it.
Thanks again!

**Comment by bengtmartensson** [comment]
Look at the signal depicted [here](https://github.com/bengtmartensson/IrScrutinizer). It is a NEC1 signal, having 0x0022 = 34 pairs = 68 durantions (flash or gap) in the intro, and 0x0002 = 2 pairs = 4 durations in the repeat. In the plot, the intro is red and the repeat blue. IrSignals may have a zero intro (for example RC5), or a zero repeat.

**Comment by bengtmartensson** [comment]
Is this still open?
